src/Pact/Eval.hs

```diff
@@ -591,16 +592,74 @@ moduleResolver lkp i mn = do
 
 
 resolveRef :: HasInfo i => i -> Name -> Eval e (Maybe Ref)
-resolveRef i (QName (QualifiedName q n _)) = moduleResolver (lookupQn n) i q
+resolveRef i (QName (QualifiedName q@(ModuleName refNs ns) n _)) = moduleResolver (lookupQn n) i q
   where
     lookupQn n' i' q' = do
       m <- lookupModule i' q'
-      return $ join $ HM.lookup n' . _mdRefMap <$> m
-resolveRef _i nn@Name {} = do
+      case m of
+        Nothing
+          | isJust ns -> evalError' i
```

Comment:
Just return `Nothing` here no? Failures of this sort simply mean that its a bad ref, not necessarily a module reference failure.

Reply:
Also, not a huge fan of casing on `isJust ...` `otherwise`. Instead, why not case on `(m,ns)`?